codecov_auth/authentication/repo_auth.py

```diff
@@ -253,86 +256,64 @@ def authenticate_credentials(self, token):
-    for the upstream repo when running in a PR.
-
-    While it uses the same "shell" (authentication.TokenAuthentication)
-    it doesn't really rely on tokens to authenticate.
-    """
-
-    auth_failed_message = "Not valid tokenless upload"
-    rate_limit_failed_message = "Tokenless has reached GitHub rate limit. Please upload using a token: https://docs.codecov.com/docs/adding-the-codecov-token."
-
-    def _get_repo_info_from_request_path(self, request) -> Repository:
+    def _get_info_from_request_path(
+        self, request
+    ) -> tuple[Repository, str | None] | None:
         path_info = request.get_full_path_info()
         # The repo part comes from https://stackoverflow.com/a/22312124
-        upload_views_prefix_regex = r"\/upload\/(\w+)\/([\w\.@:_/\-~]+)\/commits"
+        upload_views_prefix_regex = (
+            r"\/upload\/(\w+)\/([\w\.@:_/\-~]+)\/commits(?:\/([a-f0-9]{40}))?"
+        )
         match = re.search(upload_views_prefix_regex, path_info)
 
         if match is None:
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
+            return None, None
+
         # Validate provider
         service = match.group(1)
         try:
             service_enum = Service(service)
-            # Currently only Github is supported
-            # TODO [codecov/engineering-team#914]: Extend tokenless support to other providers
-            if service_enum != Service.GITHUB:
-                raise exceptions.AuthenticationFailed(self.auth_failed_message)
         except ValueError:
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
+            return None, None
+
         # Validate that next group exists and decode slug
         encoded_slug = match.group(2)
         repo = get_repository_from_string(service_enum, encoded_slug)
+
         if repo is None:
             # Purposefully using the generic message so that we don't tell that
             # we don't have a certain repo
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
-        return repo
+            return None, None
 
-    @async_to_sync
-    async def get_pull_request_info(self, repository_service, fork_pr: str):
-        try:
-            return await repository_service.get_pull_request(fork_pr)
-        except TorngitObjectNotFoundError:
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
-        except TorngitRateLimitError as e:
-            metrics.incr("auth.get_pr_info.rate_limit_hit")
-            sentry_metrics.incr("auth.get_pr_info.rate_limit_hit")
-            if e.reset:
-                now_timestamp = datetime.now().timestamp()
-                retry_after = int(e.reset) - int(now_timestamp)
-            elif e.retry_after:
-                retry_after = int(e.retry_after)
-            raise exceptions.Throttled(
-                wait=retry_after, detail=self.rate_limit_failed_message
-            )
+        commitid = match.group(3)
```

Comment:
I wonder if it would be better to aggregate the `match.group(x)` calls into one paragraph so it's easier to see